src/xdocs/checks/blocks/rightcurly.xml

```diff
@@ -113,43 +113,52 @@
         </source>
         <p id="Example1-code">Example:</p>
         <source>
-public class Test {
+public class Example1 {
 
   public void test() {
 
+    boolean foo = false;
     if (foo) {
       bar();
-    }           // violation, right curly must be in the same line as the 'else' keyword
+    } // violation, ''}' at column 5 should be on the same line'
+    // 'as the next part of a multi-block statement (one that directly'
+    // 'contains multiple blocks: if/else-if/else, do/while or try/catch/finally).'
     else {
       bar();
     }
 
     if (foo) {
       bar();
-    } else {     // OK
+    } else {
       bar();
     }
 
-    if (foo) { bar(); } int i = 0;  // violation
-                  // ^^^ statement is not allowed on same line after curly right brace
+    if (foo) { bar(); } int i = 0;
+    // violation above, ''}' at column 23 should be alone on a line.'
 
-    if (foo) { bar(); }            // OK
-    int i = 0;
+    if (foo) { bar(); }
+    i = 0;
 
     try {
       bar();
-    }           // violation, rightCurly must be in the same line as 'catch' keyword
+    } // violation, ''}' at column 5 should be on the same line'
+    // 'as the next part of a multi-block statement'
+    // '(one that directly contains multiple blocks: if/else-if/else,'
```

Comment:
No need to wrap all lines in `'`, it is required only for violation message, all other lines that provide some explanation should not be wrapped to single quote

Reply:
> No need to wrap all lines in `'`, it is required only for violation message, all other lines that provide some explanation should not be wrapped to single quote

regex doesnt matched if i dont wrapped it inside ' ', it is not even considering this as violation, so this is the only way i fixed it,

Reply:
done @romani